### DISCUSSION TITLE
Backend db store options?

### DISCUSSION THREAD

**Discussion by jmergy** [PhilanthropyDataCommons/service#86] [General]
Hi Team - thanks for all the awesome work on this. Just to clarify - the options for the backend db on this are very flexible. Can you provide some comments around that for those that may have concerns the backend db is limited to specific tech? Thanks @slifty

**Comment by slifty** [comment]
Hello!  So this is a project that uses Postgres, which is a popular (powerful!) open source SQL implementation.  It's possible to run postgres in a wide variety of environments -- you can install it directly to a server of course but also Amazon RDS supports postgres and there are well maintained Docker / Kubernetes containers as well.

Other flavors of SQL would not be supported out of the box (e.g. MySQL, Microsoft SQL, etc).  This is because they would require us to swap out libraries and also update our error processing logic throughout the endpoints (for instance we have code to map various postgres error codes to various HTTP error codes).

I hope that answers the core question.

**Reply by jmergy** [reply]
It does. If someone wanted to run something different than Postgres, they could fork and help build support for that?

**Reply by slifty** [reply]
That is correct!

**Reply by kfogel** [reply]
I'm assuming that "fork" means "clone to contribute" in this context -- a typical GitHub-style fork, that is.

We would happily accept contributions to support other database back ends, and it's not uncommon for projects (e.g., MediaWiki, for one; Django for another I think) to support different DBs.  PostgreSQL is definitely the right choice for the first one supported, though: it's very widely used, rock solid, easy to develop for, etc.

**Reply by jmergy** [reply]
Thanks @kfogel yes exactly just wanted to mention that for folks if they concerned about that. Other DBs are not being actively developed, but they could if someone was interested and willing to contribute to the project for that purpose.
